setup.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+###############################################################################
+# Bash script that downloads and does setup for a Dart SDK.                   #
+# Takes three params; first listed is the default:                            #
+# $1: Dart channel: stable|beta|dev                                           #
+# $2: OS: Linux|Windows|macOS                                                 #
+# $3: ARCH: x64|ia32                                                          #
+###############################################################################
+
+
+# Parse args.
+CHANNEL="${1:-stable}"
+OS="${2:-Linux}"
+ARCH="${3:-x64}"
+OS=$(echo "$OS" | awk '{print tolower($0)}')
+echo "Installing Dart SDK from the ${CHANNEL} channel on ${OS}-${ARCH}"
+
+# Calculate download Url. Based on:
+# https://dart.dev/tools/sdk/archive#download-urls
+PREFIX="https://storage.googleapis.com/dart-archive/channels"
+URL="${PREFIX}/${CHANNEL}/release/latest/sdk/dartsdk-${OS}-${ARCH}-release.zip"
+echo "Downloading ${URL}..."
+
+# Download installation zip.
+curl --connect-timeout 15 --retry 5 "$URL" > "${HOME}/dartsdk.zip"
+unzip "${HOME}/dartsdk.zip" -d "${RUNNER_TOOL_CACHE}" > /dev/null
```

Comment:
Perhaps we should download this into ${RUNNER_TOOL_CACHE}/dart/<version>/

That would enable checking if the version is in the cache later. It would mean we'd have to resolve `latest` to a specific version though.

Reply:
Opened https://github.com/dart-lang/setup-dart/issues/2 for that, will take a look at that later